exchange/exchange.go

```diff
@@ -232,7 +232,7 @@ func errorsToMetric(errs []error) map[pbsmetrics.AdapterError]struct{} {
 	ret := make(map[pbsmetrics.AdapterError]struct{}, len(errs))
 	var s struct{}
 	for _, err := range errs {
-		if err == context.DeadlineExceeded {
+		if _, ok := err.(*errortypes.Timeout); ok {
```

Comment:
This should be using the `DecodeError` method now.

Reply:
What do you mean?